src/app/models/form.server.model.ts

```diff
@@ -586,6 +586,37 @@ const compileFormModel = (db: Mongoose): IFormModel => {
       return next(err)
     }
 
+    // Check if MyInfo fields exist and if they do, the total number of fields <= 30, and that
+    // the response mode and authType fields are set appropriately.
+    let myInfoFieldCount = 0
+    if (this.form_fields !== undefined) {
+      this.form_fields.forEach((field) => {
+        if (field.myInfo !== undefined) {
+          myInfoFieldCount++
+        }
+      })
+    }
+
+    if (
+      myInfoFieldCount > 0 &&
+      (this.authType !== AuthType.MyInfo ||
+        this.responseMode !== ResponseMode.Encrypt)
```

Comment:
I think this needs to be flipped
```suggestion
        this.responseMode === ResponseMode.Encrypt)
```